src/c++/perf_analyzer/load_manager.cc

```diff
@@ -166,6 +166,16 @@ LoadManager::SwapTimestamps(TimestampVector& new_timestamps)
   return cb::Error::Success;
 }
 
+uint64_t
+LoadManager::CountCollectedRequests()
+{
+  uint64_t num_of_requests = 0;
+  for (auto& thread_stat : threads_stat_) {
+    num_of_requests += thread_stat->request_timestamps_.size();
```

Comment:
I think we still need to protect the request_timestamps_ vector within a mutex. You can thread_stat specific lock when calling `.size()`. 

> The container is accessed.
> No contained elements are accessed: concurrently accessing or modifying them is safe.

Source: https://www.cplusplus.com/reference/vector/vector/size/ 

It does looks like a read-only operation and not modifying the vector contents. Assume calling `.size()` when an element is being added to the container. The value returned by `size()` is undefined.

Reply:
Good catch! Fixed.